# Cylinder bandy

http://archive.ics.uci.edu/dataset/32/cylinder+bands

In [35]:
import pandas as pd
import numpy as np
from scipy.io import savemat, loadmat
path = 'Datasets/Article2022_Marcelino_Pedreira_FSP/'
df = pd.read_csv(path +'cylinder+bands/bands.data.txt', header=None)
df = df.astype('category')
df.head()

0     1        2      3    4    5    6       7         8         9   \
0  19910108  X126  TVGUIDE  25503  YES  KEY  YES  BENTON  GALLATIN  UNCOATED   
1  19910109  X266  TVGUIDE  25503  YES  KEY  YES  BENTON  GALLATIN  UNCOATED   
2  19910104    B7   MODMAT  47201  YES  KEY  YES  BENTON  GALLATIN  UNCOATED   
3  19910104  T133   MASSEY  39039  YES  KEY  YES  BENTON  GALLATIN  UNCOATED   
4  19910111   J34    KMART  37351   NO  KEY  YES  BENTON  GALLATIN  UNCOATED   

   ...    30 31 32   33   34  35  36      37   38      39  
0  ...  36.4  0  0  2.5    1  34  40     105  100    band  
1  ...  38.5  0  0  2.5  0.7  34  40     105  100  noband  
2  ...  39.8  0  0  2.8  0.9  40  40  103.87  100  noband  
3  ...  38.8  0  0  2.5  1.3  40  40  108.06  100  noband  
4  ...  42.5  5  0  2.3  0.6  35  40  106.67  100  noband  

[5 rows x 40 columns]

Attribute information given in the file "bands.names"


| Attribute        | Domain                             |
|------------------|------------------------------------|
| timestamp        | numeric; 19500101 - 21001231       |
| job number       | nominal;                           |
| press            | nominal; 821, 802, 813, 824, 815, 816, 827, 828 |
| unit number      | nominal; 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 |
| plating tank     | nominal; 1910, 1911, other         |
| proof cut        | numeric; 0 - 100                   |
| viscosity        | numeric; 0 - 100                   |
| caliper          | numeric; 0 - 1                     |
| ink temperature  | numeric; 5 - 30                    |
| humifity         | numeric; 5 - 120                   |
| roughness        | numeric; 0 - 2                     |
| blade pressure   | numeric; 10 - 75                   |
| varnish pct      | numeric; 0 - 100                   |
| press speed      | numeric; 0 - 4000                  |
| ink pct          | numeric; 0 - 100                   |
| solvent pct      | numeric; 0 - 100                   |
| ESA Voltage      | numeric; 0 - 16                    |
| ESA Amperage     | numeric; 0 - 10                    |
| wax              | numeric; 0 - 4                     |
| hardener         | numeric; 0 - 3                     |
| roller durometer | numeric; 15 - 120                  |
| current density  | numeric; 20 - 50                   |
| anode space ratio| numeric; 70 - 130                  |
| chrome content   | numeric; 80 - 120                  |
| band type        | nominal; class; band, no band      |


We will utilize numerical attributes  $21$ to $39$ to construct matrix $X$. In column $40$, we have the class label, which will be used to define vector $y$. Here is a summary of these variables.

Cleaning data

In [36]:
df = df.iloc[:, 20:40]
df = df.dropna()
df = df.reset_index(drop=True)
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)



In [3]:
x = df.iloc[:, 0:19].values
y = df.iloc[:, -1].values


y, _ = pd.factorize(y)
y = y.reshape(-1, 1)

print(y)
savemat(path + '/python/bands.mat', {'x': x, 'y': y})
dfNew = pd.DataFrame(np.concatenate((x, y), axis=1))
dfNew.to_csv(path + '/python/bands.csv', index=False, header=False)

[[0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [0]
 [1]
 [1]
 [1]
 [0]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [0]
 [1]
 [0]
 [1]
 [1]
 [0]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]


# Bupa Liver Disorders

https://archive.ics.uci.edu/dataset/60/liver+disorders

In [4]:
df = pd.read_csv(path + '/liver+disorders/bupa.data.txt', header=None)
df = df.astype('category')
df.head()


0   1   2   3   4    5  6
0  85  92  45  27  31  0.0  1
1  85  64  59  32  23  0.0  2
2  86  54  33  16  54  0.0  2
3  91  78  34  24  36  0.0  2
4  87  70  12  28  10  0.0  2

| Attribute | Description                                      |
|-----------|--------------------------------------------------|
| mcv       | mean corpuscular volume                          |
| alkphos   | alkaline phosphotase                             |
| sgpt      | alamine aminotransferase                         |
| sgot      | aspartate aminotransferase                       |
| gammagt   | gamma-glutamyl transpeptidase                    |
| drinks    | number of half-pint equivalents of alcoholic beverages consumed per day |
| selector  | field used to split data into two                |


In [5]:
df = df.iloc[:, 0:6]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)


In [6]:
x = df.iloc[:, 0:5].values
print(x)
y = df.iloc[:, -1].values
print(y)

[[85 92 45 27 31]
 [85 64 59 32 23]
 [86 54 33 16 54]
 ...
 [98 77 55 35 89]
 [91 68 27 26 14]
 [98 99 57 45 65]]
[0.0, 0.0, 0.0, 0.0, 0.0, ..., 12.0, 12.0, 15.0, 16.0, 20.0]
Length: 345
Categories (16, float64): [0.0, 0.5, 1.0, 2.0, ..., 12.0, 15.0, 16.0, 20.0]


In [7]:

yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] < 3:
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)



savemat(path + '/python/bupa.mat', {'x': x, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((x, yNew), axis=1))
dfNew.to_csv(path + '/python/bupa.csv', index=False, header=False)


[[1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]


# Contraceptive Method Choice
http://archive.ics.uci.edu/dataset/30/contraceptive+method+choice

In [8]:
df = pd.read_csv(path + 'contraceptive+method+choice/cmc.data.txt', header=None)
df = df.astype('category')
df.head()

0  1  2   3  4  5  6  7  8  9
0  24  2  3   3  1  1  2  3  0  1
1  45  1  3  10  1  1  3  4  0  1
2  43  2  3   7  1  1  3  4  0  1
3  42  3  2   9  1  1  3  3  0  1
4  36  3  3   8  1  1  3  2  0  1

| Attribute | Description | Domain |
|-----------|-------------|--------|
| Wife's age | numerical |  |
| Wife's education | categorical | 1=low, 2, 3, 4=high |
| Husband's education | categorical | 1=low, 2, 3, 4=high |
| Number of children ever born | numerical |  |
| Wife's religion | binary | 0=Non-Islam, 1=Islam |
| Wife's now working? | binary | 0=Yes, 1=No |
| Husband's occupation | categorical | 1, 2, 3, 4 |
| Standard-of-living index | categorical | 1=low, 2, 3, 4=high |
| Media exposure | binary | 0=Good, 1=Not good |
| Contraceptive method used | class attribute | 1=No-use 2=Long-term 3=Short-term |


Cleaning data   

In [9]:
df = df.iloc[:, 0:10]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)

Splitting data into $X$ and $y$

In [10]:
x = df.iloc[:, 0:9].values
y = df.iloc[:, -1].values
print(y)

[1, 1, 1, 1, 1, ..., 3, 3, 3, 3, 3]
Length: 1473
Categories (3, int64): [1, 2, 3]


Convert the categorical $y$ data in  into numerical. The equivalence is as folows.

    1.Keep the categorical 1
    2.The categoricals 2 and 3 are converted to 2

In [11]:
for i in range(len(y)):
    if y[i] == 1:
        y[i] = 1
    else:
        y[i] = 2
  
y, _ = pd.factorize(y)
y = y.reshape(-1, 1)
print(y)

savemat(path + '/python/cmc.mat', {'x': x, 'y': y})
dfNew = pd.DataFrame(np.concatenate((x, y), axis=1))
dfNew.to_csv(path + '/python/cmc.csv', index=False, header=False)

[[0]
 [0]
 [0]
 ...
 [1]
 [1]
 [1]]


# Haberman's Survival

http://archive.ics.uci.edu/dataset/43/haberman+s+survival

In [12]:
df = pd.read_csv(path + 'haberman+s+survival/haberman.data.txt', header=None)
df = df.astype('category')
df.head()

0   1  2  3
0  30  64  1  1
1  30  62  3  1
2  30  65  0  1
3  31  59  2  1
4  31  65  4  1

Atributte information:

| No. | Feature                                     | Description                               |
| --- | ------------------------------------------- | ----------------------------------------- |
| 1   | Age of patient                              | Numerical value indicating age at operation |
| 2   | Year of operation                           | Numerical value indicating year - 1900     |
| 3   | Number of positive axillary nodes detected  | Numerical value indicating node count      |
| 4   | Survival status                             | Class attribute: 1 = survived 5 years or longer, 2 = died within 5 years |

Cleaning data:

In [13]:
df = df.iloc[:, 0:4]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)


Splitting data into $X$ and $y$


In [14]:
x = df.iloc[:, 0:3].values
y = df.iloc[:, -1].values

yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] == 1:
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)

savemat(path + '/python/haberman.mat', {'x': x, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((x, yNew), axis=1))
dfNew.to_csv(path + '/python/haberman.csv', index=False, header=False)

[[1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]


# Heart Disease

http://archive.ics.uci.edu/dataset/145/statlog+heart (Arquivo vazio)
https://sci2s.ugr.es/keel/dataset.php?cod=99

In [15]:
df = pd.read_csv(path + 'heart+disease/heart.dat.txt', header=None)
df = df.astype('category')
df.head()


0  1  2    3    4  5  6    7  8     9  10 11 12 13
0  70  1  4  130  322  0  2  109  0  24.0  2  3  3  2
1  67  0  3  115  564  0  2  160  0  16.0  2  0  7  1
2  57  1  2  124  261  0  0  141  0   3.0  1  0  7  2
3  64  1  4  128  263  0  0  105  1   2.0  2  1  7  1
4  74  0  2  120  269  0  2  121  1   2.0  1  1  3  1

Atributte Information about heart.dat.txt


| Attribute                   | Domain        |
|-----------------------------|---------------|
| Age                         | [29, 77]      |
| Sex                         | [0, 1]        |
| ChestPainType               | [1, 4]        |
| RestBloodPressure           | [94, 200]     |
| SerumCholestoral            | [126, 564]    |
| FastingBloodSugar           | [0, 1]        |
| ResElectrocardiographic    | [0, 2]        |
| MaxHeartRate                | [71, 202]     |
| ExerciseInduced             | [0, 1]        |
| Oldpeak                     | [0.0, 62.0]   |
| Slope                       | [1, 3]        |
| MajorVessels                | [0, 3]        |
| Thal                        | [3, 7]        |
| Class                       | {1, 2}        |


Cleaning data:

In [16]:
df = df.iloc[:, 0:14]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)


Splitting data into $X$ and $y$

In [17]:
X = df.iloc[:, 0:13].values
y = df.iloc[:, -1].values

yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] == 1:
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)


[[2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [2]
 [1]
 [2]
 [2]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]


In [18]:
savemat(path + '/python/heart.mat', {'x': X, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((X, yNew), axis=1))
dfNew.to_csv(path + '/python/heart.csv', index=False, header=False)


# Hepatitis

http://archive.ics.uci.edu/dataset/46/hepatitis

In [19]:
df = pd.read_csv(path + 'hepatitis/hepatitis.data.txt', header=None)
df = df.astype('category')
df.head()

0   1  2  3  4  5  6  7  8  9  10 11 12 13    14   15   16   17  18 19
0  2  30  2  1  2  2  2  2  1  2  2  2  2  2  1.00   85   18  4.0   ?  1
1  2  50  1  1  2  1  2  2  1  2  2  2  2  2  0.90  135   42  3.5   ?  1
2  2  78  1  2  2  1  2  2  2  2  2  2  2  2  0.70   96   32  4.0   ?  1
3  2  31  1  ?  1  2  2  2  2  2  2  2  2  2  0.70   46   52  4.0  80  1
4  2  34  1  2  2  2  2  2  2  2  2  2  2  2  1.00    ?  200  4.0   ?  1

| Attribute       | Domain                     |
|-----------------|----------------------------|
| Class           | DIE, LIVE                  |
| AGE             | 10, 20, 30, 40, 50, 60, 70, 80 |
| SEX             | male, female               |
| STEROID         | no, yes                    |
| ANTIVIRALS      | no, yes                    |
| FATIGUE         | no, yes                    |
| MALAISE         | no, yes                    |
| ANOREXIA        | no, yes                    |
| LIVER BIG       | no, yes                    |
| LIVER FIRM      | no, yes                    |
| SPLEEN PALPABLE | no, yes                    |
| SPIDERS         | no, yes                    |
| ASCITES         | no, yes                    |
| VARICES         | no, yes                    |
| BILIRUBIN       | 0.39, 0.80, 1.20, 2.00, 3.00, 4.00 |
| ALK PHOSPHATE   | 33, 80, 120, 160, 200, 250 |
| SGOT            | 13, 100, 200, 300, 400, 500 |
| ALBUMIN         | 2.1, 3.0, 3.8, 4.5, 5.0, 6.0 |
| PROTIME         | 10, 20, 30, 40, 50, 60, 70, 80, 90 |
| HISTOLOGY       | no, yes                    |


Cleaning data:

In [20]:

df = df.iloc[:, 0:20]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)


Splitting data into $X$ and $y$

In [21]:
X = df.iloc[:, 1:20].values 
y = df.iloc[:, 0].values

yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] == 1:
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)

[[2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [2]
 [2]
 [1]
 [1]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [2]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]]


In [22]:

savemat(path + '/python/hepatitis.mat', {'x': X, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((X, yNew), axis=1))
dfNew.to_csv(path + '/python/hepatitis.csv', index=False, header=False)


# Pima      

https://www.kaggle.com/datasets/uciml/pima-indians-diabetes-database

In [23]:
df = pd.read_csv(path + 'pima/diabetes.csv', header=None)
df = df.astype('category')
df.head()

0        1              2              3        4     5  \
0  Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI   
1            6      148             72             35        0  33.6   
2            1       85             66             29        0  26.6   
3            8      183             64              0        0  23.3   
4            1       89             66             23       94  28.1   

                          6    7        8  
0  DiabetesPedigreeFunction  Age  Outcome  
1                     0.627   50        1  
2                     0.351   31        0  
3                     0.672   32        1  
4                     0.167   21        0

| No. | Feature                    | Description                                      |
| --- | -------------------------- | ------------------------------------------------ |
| 1   | Pregnancies                | Number of times pregnant                         |
| 2   | Glucose                    | Plasma glucose concentration at 2 hours          |
| 3   | BloodPressure              | Diastolic blood pressure (mm Hg)                 |
| 4   | SkinThickness              | Triceps skin fold thickness (mm)                 |
| 5   | Insulin                    | 2-Hour serum insulin (mu U/ml)                   |
| 6   | BMI                        | Body mass index (weight in kg/(height in m)^2)   |
| 7   | DiabetesPedigreeFunction   | Diabetes pedigree function                       |
| 8   | Age                        | Age (years)                                      |
| 9   | Outcome                    | Class variable: 0 or 1 (1 represents diabetes)    |
|     |                            | 268 of 768 instances are 1, the others are 0      |

Cleaning data:

In [24]:
#Remover a primeira linha
df = df.iloc[1:, :]
df = df.iloc[:, 0:9]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)


Splitting data into $X$ and $y$

In [25]:
X = df.iloc[:, 0:8].values
y = df.iloc[:, -1].values
print(y)
yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] == '0':
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)


['1', '0', '1', '0', '1', ..., '0', '0', '0', '1', '0']
Length: 768
Categories (3, object): ['0', '1', 'Outcome']
[[2]
 [1]
 [2]
 [1]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [2]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [1]
 [1]
 [1]
 [2]
 [1]
 

In [26]:

savemat(path + '/python/pima.mat', {'x': X, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((X, yNew), axis=1))
dfNew.to_csv(path + '/python/pima.csv', index=False, header=False)


# Sonar 

http://archive.ics.uci.edu/dataset/151/connectionist+bench+sonar+mines+vs+rocks

In [27]:
df = pd.read_csv(path + 'sonar/sonar.all-data.csv', header=None)
df = df.astype('category')
df.head()


0       1       2       3       4       5       6       7       8   \
0  Freq_1  Freq_2  Freq_3  Freq_4  Freq_5  Freq_6  Freq_7  Freq_8  Freq_9   
1    0.02  0.0371  0.0428  0.0207  0.0954  0.0986  0.1539  0.1601  0.3109   
2  0.0453  0.0523  0.0843  0.0689  0.1183  0.2583  0.2156  0.3481  0.3337   
3  0.0262  0.0582  0.1099  0.1083  0.0974   0.228  0.2431  0.3771  0.5598   
4    0.01  0.0171  0.0623  0.0205  0.0205  0.0368  0.1098  0.1276  0.0598   

        9   ...       51       52       53       54       55       56  \
0  Freq_10  ...  Freq_52  Freq_53  Freq_54  Freq_55  Freq_56  Freq_57   
1   0.2111  ...   0.0027   0.0065   0.0159   0.0072   0.0167    0.018   
2   0.2872  ...   0.0084   0.0089   0.0048   0.0094   0.0191    0.014   
3   0.6194  ...   0.0232   0.0166   0.0095    0.018   0.0244   0.0316   
4   0.1264  ...   0.0121   0.0036    0.015   0.0085   0.0073    0.005   

        57       58       59     60  
0  Freq_58  Freq_59  Freq_60  Label  
1   0.0084    0.009   0.0032      R  
2   0.0049   0.0052   0.0044      R  
3   0.0164   0.0095   0.0078      R  
4   0.0044    0.004   0.0117      R  

[5 rows x 61 columns]

Features are all continuous and are all about frequency response of sonar chirp returns bouncing off different surfaces. The label associated with each record contains the letter R if the object is a rock and M if it is a mine (metal cylinder).

In [28]:
#Remover a primeira linha
df = df.iloc[1:, :]
df = df.iloc[:, 0:61]
df = df.dropna()
#Remove '?' from the dataset
df = df[(df != '?').all(axis=1)]
df = df.reset_index(drop=True)
print(df.head())

       0       1       2       3       4       5       6       7       8   \
0    0.02  0.0371  0.0428  0.0207  0.0954  0.0986  0.1539  0.1601  0.3109   
1  0.0453  0.0523  0.0843  0.0689  0.1183  0.2583  0.2156  0.3481  0.3337   
2  0.0262  0.0582  0.1099  0.1083  0.0974   0.228  0.2431  0.3771  0.5598   
3    0.01  0.0171  0.0623  0.0205  0.0205  0.0368  0.1098  0.1276  0.0598   
4  0.0762  0.0666  0.0481  0.0394   0.059  0.0649  0.1209  0.2467  0.3564   

       9   ...      51      52      53      54      55      56      57  \
0  0.2111  ...  0.0027  0.0065  0.0159  0.0072  0.0167   0.018  0.0084   
1  0.2872  ...  0.0084  0.0089  0.0048  0.0094  0.0191   0.014  0.0049   
2  0.6194  ...  0.0232  0.0166  0.0095   0.018  0.0244  0.0316  0.0164   
3  0.1264  ...  0.0121  0.0036   0.015  0.0085  0.0073   0.005  0.0044   
4  0.4459  ...  0.0031  0.0054  0.0105   0.011  0.0015  0.0072  0.0048   

       58      59 60  
0   0.009  0.0032  R  
1  0.0052  0.0044  R  
2  0.0095  0.0078  R  


Splitting data into $X$ and $y$

Y follows the following equivalence:

    1.Keep the categorical R
    2.The categoricals M are converted to 2

In [29]:
X = df.iloc[:, 0:60].values
y = df.iloc[:, -1].values
print(y)
yNew = np.zeros((len(y), 1))
for i in range(len(y)):
    if y[i] == 'R':
        yNew[i] = 1
    else:
        yNew[i] = 2


#Converter para int
yNew = yNew.astype(int)
print(yNew)


['R', 'R', 'R', 'R', 'R', ..., 'M', 'M', 'M', 'M', 'M']
Length: 208
Categories (3, object): ['Label', 'M', 'R']
[[1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [1]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2]
 [2

In [30]:

savemat(path + '/python/sonar.mat', {'x': X, 'y': yNew})
dfNew = pd.DataFrame(np.concatenate((X, yNew), axis=1))
dfNew.to_csv(path + '/python/sonar.csv', index=False, header=False)